src/django_components/dependencies.py

```diff
@@ -917,37 +957,50 @@ def _insert_js_css_to_default_locations(
     This function tries to insert the JS and CSS content into the default locations.
 
     JS is inserted at the end of `<body>`, and CSS is inserted at the end of `<head>`.
-    """
-    elems = SoupNode.from_fragment(html_content)
 
-    if not elems:
+    We find these tags by looking for the first `</head>` and last `</body>` tags.
```

Comment:
And this section is about inserting JS and CSS into their default locations.

Now, to avoid having to parse the HTML to find the end of body and head, we use following approach:
  - We find the end of `<body>` by searching for **last** occurrence of `</body>`
  - And for the end of `<head>` we search for the **first** occurrence of `</head>`